Kommunicate/Classes/KMConversationListViewController.swift

```diff
@@ -0,0 +1,642 @@
+            dismiss(animated: true, completion: nil)
+        }
+    }
+
+    override public func showAccountSuspensionView() {
+        let accountVC = ALKAccountSuspensionController()
+        present(accountVC, animated: false, completion: nil)
+        accountVC.closePressed = { [weak self] in
+            let popVC = self?.navigationController?.popViewController(animated: true)
+            if popVC == nil {
+                self?.navigationController?.dismiss(animated: true, completion: nil)
+            }
+        }
+    }
+
+    func conversationVC() -> KMConversationViewController? {
+        return navigationController?.topViewController as? KMConversationViewController
+    }
+
+    fileprivate func push(conversationVC: KMConversationViewController, with viewModel: ALKConversationViewModel) {
+        if let topVC = navigationController?.topViewController as? KMConversationViewController {
+            // Update the details and refresh
+            topVC.unsubscribingChannel()
+            topVC.viewModel.contactId = viewModel.contactId
+            topVC.viewModel.channelKey = viewModel.channelKey
+            topVC.viewModel.conversationProxy = viewModel.conversationProxy
+            topVC.viewWillLoadFromTappingOnNotification()
+            topVC.refreshViewController()
+        } else {
+            // push conversation VC
+            conversationVC.viewWillLoadFromTappingOnNotification()
+            navigationController?.pushViewController(conversationVC, animated: false)
+        }
+    }
+
+    private func showAlert(viewController:KMConversationListViewController) {
+        let alertMessage = LocalizedText.unableToCreateConversationError
+        let okText = LocalizedText.okButton
+        let alert = UIAlertController(
+            title: "",
+            message: alertMessage,
+            preferredStyle: UIAlertController.Style.alert)
+        alert.addAction(UIAlertAction(title: okText, style: UIAlertAction.Style.default, handler: nil))
+        viewController.present(alert, animated: true, completion: nil)
+    }
+
+    private func createConversationAndLaunch(notification:Notification){
+
+        guard let vc = notification.object as? KMConversationListViewController else {
+            return
+        }
+        vc.view.isUserInteractionEnabled = false
+        vc.navigationController?.view.isUserInteractionEnabled = false
+        let alertView =  displayAlert(viewController : vc)
+
+        Kommunicate.createConversation() { (result) in
+            switch result {
+            case .success(let conversationId):
+                DispatchQueue.main.async {
+                    vc.view.isUserInteractionEnabled = true
+                    vc.navigationController?.view.isUserInteractionEnabled = true
+                    alertView.dismiss(animated: false, completion: nil)
+                    Kommunicate.showConversationWith(groupId: conversationId, from: vc, completionHandler: { (success) in
```

Comment:
Instead of using `Kommunicate`'s method to launch, we can use what we have defined on top: `launchChat(groupId:)`. As we use the same when the user taps on a conversation.

Reply:
this is pending

Reply:
the return type is conversationId which is clientGroupId and method ```launchChat(groupId:)``` it takes an int groupId, this is the only method we have in Kommunicate with ```clientGroupId```
```Kommunicate.showConversationWith( ```

Reply:
Then we can get the `groupId` through channel info API and pass the same to `launchChat()`.